packages/flutter_tools/lib/src/windows/visual_studio.dart

```diff
@@ -127,11 +127,11 @@ class VisualStudio {
     // component name is significantly different in different versions.
     // Default to the latest known description, but use a specific string
     // if a known older version is requested.
-    String cppToolchainDescription = 'MSVC v142 - VS 2019 C++ x64/x86 build tools (v14.21)';
+    String cppToolchainDescription = 'MSVC v142 - VS 2019 C++ x64/x86 build tools';
     if (visualStudioMajorVersion == 15) {
-      cppToolchainDescription = 'VC++ 2017 version 15.9 v14.16 latest v141 tools';
+      cppToolchainDescription = 'VC++ 2017 version 15.9 v14.## latest v141 tools';
     }
-
+    cppToolchainDescription += '\n - If there are multiple versions, install the latest one';
```

Comment:
Since the reason for this is non-obvious for someone who hasn't been dealing with the related bugs, let's add a comment on this line. How about:
```
// The 'Microsoft.VisualStudio.Component.VC.Tools.x86.x64' ID is assigned to the latest
// release of the toolchain, and there can be minor updates within a given version of
// Visual Studio. Since it changes over time, listing a precise version would become
// wrong after each VC++ toolchain update, so just instruct people to install the
// latest version.
```